scipy/stats/_fit.py

```diff
@@ -1254,6 +1270,27 @@ def _filliben(dist, data):
 _filliben.alternative = 'less'  # type: ignore[attr-defined]
 
 
+def _order_statistic_evs(dist, n):
+    # Calculate the expected values of the order statistics of a distribution
+    # PDF from https://en.wikipedia.org/wiki/Order_statistic
+    r = np.arange(1, n + 1)
+    constant = gammaln(n + 1) - gammaln(r) - gammaln(n - r + 1)
+    def integrand(x):
+        variable = dist.logpdf(x) + (r-1)*dist.logcdf(x) + (n-r)*dist.logsf(x)
+        return x * np.exp(constant + variable)
+
+    res = integrate.quad_vec(integrand, -np.inf, np.inf)
+    return res[0]
```

Comment:
![image](https://user-images.githubusercontent.com/6570539/224583234-d320679c-2258-42f1-ba86-78e68fa01bf1.png)

The code follows from $\log(a/b) = \log(a) - \log(b)$ and $\Gamma(n) = (n-1)!$. As much of the calculation is done in the logspace as possible to avoid premature underflows when raising small CDFs/SFs to powers.